test/e2e/fishapp/shared/manifest_builder.go

```diff
@@ -0,0 +1,275 @@
+package shared
+
+import (
+	"fmt"
+	appmeshv1beta1 "github.com/aws/aws-app-mesh-controller-for-k8s/pkg/apis/appmesh/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	defaultAppImage       = "970805265562.dkr.ecr.us-west-2.amazonaws.com/colorteller:latest"
+	defaultHTTPProxyImage = "abhinavsingh/proxy.py:latest"
+)
+
+const (
+	appContainerPort       = 9080
+	httpProxyContainerPort = 8899
+)
+
+type ServiceDiscoveryType int
+
+const (
+	DNSServiceDiscovery ServiceDiscoveryType = iota
+	CloudMapServiceDiscovery
+)
+
+type ManifestBuilder struct {
+	MeshName             string
+	Namespace            string
+	ServiceDiscoveryType ServiceDiscoveryType
+
+	// required when serviceDiscoveryType == CloudMapServiceDiscovery
+	CloudMapNamespace string
+}
+
+func (b *ManifestBuilder) BuildNodeDeployment(instanceName string, replicas int32) *appsv1.Deployment {
```

Comment:
will this rely on appmesh-injector running or should we remove that dependency?

Reply:
yes, we rely on appmesh-injector. I don't think we should remove it, since the test is meant to test them as whole. BTW, we'll need controller when we test injector. 


Here is the plan to running test in CI for controller:
1. compile the controller and upload to ECR as controller: xxxxxx
2. create cluster with controller installed with image(controller: xxxxxx), and `latest injector` from helm chart. (with the help of this PR: https://github.com/aws/aws-k8s-tester/pull/81), which allows to install controller & injector from latest helm chart, while allows to override controller/injector image.
3. launch the test suit against the cluster
Here is the plan to running test in CI for injector:
1. compile the injector and upload to ECR as injector: xxxxxx
2. create cluster with injector installed with image(injector: xxxxxx), and `latest controller` from helm chart. (with the help of this PR: https://github.com/aws/aws-k8s-tester/pull/81), which allows to install controller & injector from latest helm chart, while allows to override controller/injector image.
3. launch the test suit against the cluster